frontend/src/metabase/admin/performance/components/StrategyEditorForDatabases.tsx

```diff
@@ -0,0 +1,328 @@
+        return;
+      }
+      const update = () => setTargetId(newTargetId);
+      if (isFormDirty) {
+        askBeforeDiscardingChanges(update);
+      } else {
+        update();
+      }
+    };
+
+    useEffect(() => {
+      if (!canOverrideRootStrategy && targetId === null) {
+        setTargetId(rootId);
+      }
+    }, [canOverrideRootStrategy, targetId]);
+
+    const saveStrategy = useCallback(
+      async (values: Strategy) => {
+        if (targetId === null) {
+          return;
+        }
+
+        const baseConfig: Pick<Config, "model" | "model_id"> = {
+          model: targetId === rootId ? "root" : "database",
+          model_id: targetId,
+        };
+
+        const otherConfigs = configs.filter(
+          config => config.model_id !== targetId,
+        );
+        if (values.type === "inherit") {
+          await CacheConfigApi.delete(baseConfig, { hasBody: true }).then(
+            () => {
+              setConfigs(otherConfigs);
+            },
+          );
+        } else {
+          const validFields = getFieldsForStrategyType(values.type);
+          const newStrategy = pick(values, validFields) as Strategy;
+
+          const validatedStrategy =
+            Strategies[values.type].validateWith.validateSync(newStrategy);
+
+          const newConfig = {
+            ...baseConfig,
+            strategy: validatedStrategy,
+          };
+
+          await CacheConfigApi.update(newConfig).then(() => {
+            setConfigs([...otherConfigs, newConfig]);
+          });
+        }
+      },
+      [configs, targetId],
+    );
+    const formPanelRef = useRef<HTMLDivElement>(null);
+    const [formPanelHasVerticalScrollbar, setFormPanelHasVerticalScrollbar] =
+      useState(false);
+
+    useEffect(() => {
+      const formPanel = formPanelRef.current;
+      if (!formPanel) {
+        return;
+      }
+      setFormPanelHasVerticalScrollbar(
+        formPanel.scrollHeight > formPanel.clientHeight,
+      );
+    }, [
+      formPanelRef.current?.scrollHeight,
+      formPanelRef.current?.clientHeight,
+      formPanelRef,
+      setFormPanelHasVerticalScrollbar,
+    ]);
+
+    const showLoadingSpinner = useDelayedLoadingSpinner();
+    const error = databasesResult.error || configsResult.error;
+    const loading = databasesResult.isLoading || configsResult.loading;
+
+    if (error || loading) {
+      return showLoadingSpinner ? (
+        <LoadingAndErrorWrapper error={error} loading={loading} />
+      ) : (
+        <></>
+      );
+    }
+
+    return (
+      <TabWrapper role="region" aria-label="Data caching settings">
+        <Stack spacing="xl" lh="1.5rem" maw="32rem" mb="1.5rem">
+          <aside>{PLUGIN_CACHING.explanation}</aside>
+        </Stack>
+        {confirmationModal}
+        <Box
+          style={{
+            display: "grid",
+            gridTemplateColumns: "minmax(5rem, 30rem) minmax(5rem, auto)",
+            overflow: "hidden",
+          }}
```

Comment:
I'm seeing a lot of using of the style prop in this PR. I think this is the flaw with mantine's style attribute props... wondering if these should be styled-components instead, but I'm personally unsure of the team's philosophy here.

Reply:
During development it's faster for me to use the style prop (though I wish there was a css-to-js converter for styles). My sense is, if it's only a few style properties, it doesn't need a separate styled component.